File: go/problems/max_product_two_items.go
package problems

func MaxProduct(nums []int) int {
	largest := 0
	secondLargest := 0
	for _, num := range nums {
		if num >= largest {
			largest, secondLargest = num, largest
		} else if num > secondLargest {
			secondLargest = num
		}
	}
	return (largest - 1) * (secondLargest - 1)
}

File: rust/src/problems/max_product_two_items.rs
// https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/
pub fn max_product(nums: Vec<i32>) -> i32 {
    let mut max: i32 = 0;
    for (i, _) in nums.iter().enumerate() {
        for j in i + 1..nums.len() {
            if (nums[i] - 1) * (nums[j] - 1) > max {
                max = (nums[i] - 1) * (nums[j] - 1);
            }
        }
    }
    return max;
}